src/audio/pipeline.c

```diff
@@ -357,9 +357,11 @@ static struct task *pipeline_task_init(struct pipeline *p, uint32_t type,
 
 	task = rzalloc(RZONE_RUNTIME, SOF_MEM_CAPS_RAM, sizeof(*task));
 
-	if (task)
-		schedule_task_init(task, type, p->ipc_pipe.priority, func, NULL,
-				   p, p->ipc_pipe.core, 0);
+	if (task && schedule_task_init(task, type, p->ipc_pipe.priority, func,
+				       NULL, p, p->ipc_pipe.core, 0) < 0) {
+		rfree(task);
+		task = NULL;
+	}
 
```

Comment:
I don't like this mix of checking a variabile and calling a function inside if. 

What about this:
`` task = rzalloc();
if (!task) 
   return task;

ret = schedule_task_init();
if (!ret) {
rfree(ask);
task = NULL;
}

return task;
```

Reply:
Yea, I agree mixed checks are bad.
I changed code to more like what you suggested, but without introducing new variable.